src/obu.c

```diff
@@ -361,7 +369,13 @@ static avifBool parseAV2SequenceHeader(avifBits * bits, avifSequenceHeader * hea
     AVIF_CHECK(parseSequenceHeaderProfile(bits, header));
 
     // See av1_read_sequence_header() in avm.
-    AVIF_CHECK(parseSequenceHeaderFeatures(bits, header));
+    AVIF_CHECK(parseSequenceHeaderFrameMaxDimensions(bits, header));
+#if CONFIG_BLOCK_256
+    if (!avifBitsRead(bits, 1)) // BLOCK_256X256
+#endif
+        avifBitsRead(bits, 1); // BLOCK_128X128
```

Comment:
Nit/optional: It is a little confusing to nest `#if` and `if` this way. We probably should duplicate some code to avoid the nesting:

```
#if CONFIG_BLOCK_256
    if (!avifBitsRead(bits, 1)) // BLOCK_256X256
        avifBitsRead(bits, 1); // BLOCK_128X128
#else
    avifBitsRead(bits, 1); // BLOCK_128X128
#endif
```

Alternatively we could consider using the solution at lines 380-387. Does any compiler warn about the empty `if` block there when `CONFIG_LR_FLEX_SYNTAX` is equal to 0?

Reply:
I need this synced internally so I will merge now but feel free to send cosmetic changes afterwards. All approaches you mentioned are fine with me.